### PR TITLE
set helm client-go user-agent

### PR DESCRIPTION
**Pull request by weiwei04**
set the correct user-agent for access log & monitor

before

```
helm/v1.8.0+$Format:%h$ (linux/amd64) kubernetes/$Format
tiller/v1.8.0+$Format:%h$ (linux/amd64) kubernetes/$Format
```

after

```
/tiller/v2.7+unreleased+g44c34fc
helm/v2.7+unreleased+g44c34fc
```

Signed-off-by: Wei Wei <weiwei.inf@gmail.com>

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -65,10 +70,43 @@ type Client struct {
...
+}
+
+var _ clientcmd.ClientConfig = &clientConfig{}
+
+func (c *clientConfig) RawConfig() (clientcmdapi.Config, error) {
+	return c.inner.RawConfig()
+}
+
+func (c *clientConfig) ClientConfig() (*rest.Config, error) {
+	fmt.Printf("FUCK clientConfig.ClientConfig()\n")
```
Comment (technosophos) [inline pkg/kube/client.go]: Fix, please. 😄  (Same with other one a bit lower in code)

**Review by technosophos — CHANGES_REQUESTED** [review]
Haven't manually tested yet. But there were a few issues with output messages.

File: pkg/kube/client.go (hunk 1 above)
Comment (weiwei04) [reply pkg/kube/client.go]: Sorry, forget to delete the debug message :stuck_out_tongue_closed_eyes:

File: cmd/tiller/tiller.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func main() {
 	flag.Parse()
 
 	if *printVersion {
-		fmt.Println(version.GetVersion())
+		fmt.Println(version.FormatVersion(version.GetVersionProto(), true))
```
Comment (mattfarina) [inline cmd/tiller/tiller.go]: With the old `version.GetVersion` function when there was no `BuildMetadata` it would just share the version number (e.g., `1.2.3`). In the new setup it will always share the `GitCommit`. Even for releases (as opposed to dev versions). Why the change?

File: cmd/helm/helm.go [hunk 3]
```diff
@@ -220,6 +221,8 @@ func configForContext(context string) (*rest.Config, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not get Kubernetes config for context %q: %s", context, err)
 	}
+	config.UserAgent = fmt.Sprintf("%s/%s",
+		os.Args[0], version.FormatVersion(version.GetVersionProto(), true))
```
Comment (mattfarina) [inline cmd/helm/helm.go]: Even for releases, won't this have the short git commit hash?

**Comment by mattfarina** [comment]
Also, this needs to be rebased.
